src/projection.js

```diff
@@ -232,3 +231,15 @@ function applyProjection(cx, cy, values, projection) {
     stream.point(x[i], y[i]);
   }
 }
+
+// When a named projection is specified, we can use its natural aspect ratio to
+// determine a good value for the projection’s height based on the desired
+// width. When we don’t have a way to know, the golden ratio is our best guess.
+export function projectionAspectRatio(projection, geometry) {
+  if (isObject(projection)) projection = projection.type;
+  if (typeof projection === "string") {
+    const {aspectRatio} = namedProjection(projection);
+    if (aspectRatio) return aspectRatio;
+  }
+  if (geometry) return golden - 1;
```

Comment:
I think this isn’t quite right… we also want to use `golden - 1` if a projection has been specified even if there is no geometry (as when using a custom projection).

Reply:
good catch; should be 
```suggestion
  if (projection !== undefined || geometry) return golden - 1;
```

(since the null projection is a projection)